backend/typescript/rest/donationsRoutes.ts

```diff
@@ -52,4 +61,70 @@ donationRouter.post("/give", async (req: any, res: any) => {
   }
 });
 
+
+// Stripe webhook to handle payment events
+donationRouter.post("/webhook", bodyParser.raw({ type: 'application/json' }), async (request: Request, res: Response) => {
+  const sig = request.headers['stripe-signature'];
+
+  if (sig === undefined) {
+    return res.status(400).send("Missing Stripe signature header");
+  }
+
+  let event;
+  try {
+    event = stripe.webhooks.constructEvent(request.body, sig, process.env.STRIPE_WEBHOOK_SECRET!);
+  } catch (error) {
+    res.status(500).json({ error: getErrorMessage(error) });
+  }
+
+  // Handle the event
```

Comment:
We're going to want the code to be a bit more modular here. Can you guys define this backend logic in a separate Stripe webhook service?

This would allow us to do things like reuse the logic in another route if needed.

In this route file, we should only be defining the actual route endpoint and which service functions are called.